db/db_impl/db_impl_open.cc

```diff
@@ -1256,53 +1269,53 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& wal_numbers,
         edit->SetLogNumber(max_wal_number + 1);
       }
     }
+
     if (status.ok()) {
       // we must mark the next log number as used, even though it's
       // not actually used. that is because VersionSet assumes
       // VersionSet::next_file_number_ always to be strictly greater than any
       // log number
       versions_->MarkFileNumberUsed(max_wal_number + 1);
 
-      autovector<ColumnFamilyData*> cfds;
-      autovector<const MutableCFOptions*> cf_opts;
-      autovector<autovector<VersionEdit*>> edit_lists;
+      if (corrupted_wal_found != nullptr && *corrupted_wal_found == true &&
+          immutable_db_options_.wal_recovery_mode ==
+              WALRecoveryMode::kPointInTimeRecovery) {
+        MoveCorruptedWalFiles(wal_numbers, corrupted_wal_number);
+        // Since last log has been deleted, no need to truncate the log.
+        truncate_last_log = false;
```

Comment:
Can we just truncate the last WAL? From the implementation of `GetLogSizeAndMaybeTruncate()`, it truncates the file to its actual size to reclaim pre-allocated spaces.

Reply:
Since there have been quite a number of comments, just want to make sure that we still consider this one. :)

Reply:
Yes, we do.  Before moving  the corrupted WAL, it truncates the last WAL to reclaim the pre allocated space before.